From 887944231c6044379462999a98899149f31645c0 Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Thu, 23 Oct 2014 09:54:09 -0400
Subject: [fs] ext4: use %pd printk specificer

Message-id: <1414058125-4183-96-git-send-email-lczerner@redhat.com>
Patchwork-id: 98196
O-Subject: [RHEL 7.1 PATCH v2 095/171] ext4: use %pd printk specificer
Bugzilla: 1150139
RH-Acked-by: Eric Sandeen <sandeen@redhat.com>

From: David Howells <dhowells@redhat.com>

BZ 1150139
https://bugzilla.redhat.com/show_bug.cgi?id=1150139

BREW 8142710
https://brewweb.devel.redhat.com/taskinfo?taskID=8142710

Upstream commit a34e15cc35c743f0e49125a4fbd22a7c55b686d8

 Use the new %pd printk() specifier in Ext4 to replace passing of
 dentry name or dentry name and name length * 2 with just passing the
 dentry.

 Signed-off-by: David Howells <dhowells@redhat.com>
 Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>
 cc: Andreas Dilger <adilger.kernel@dilger.ca>
 cc: linux-ext4@vger.kernel.org
 Signed-off-by: Lukas Czerner <lczerner@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/ext4/namei.c b/fs/ext4/namei.c
index ca8174b..b855f2a 100644
--- a/fs/ext4/namei.c
+++ b/fs/ext4/namei.c
@@ -1425,9 +1425,8 @@ static struct dentry *ext4_lookup(struct inode *dir, struct dentry *dentry, unsi
    return ERR_PTR(-EIO);
   }
   if (unlikely(ino == dir->i_ino)) {
-   EXT4_ERROR_INODE(dir, "'%.*s' linked to parent dir",
-      dentry->d_name.len,
-      dentry->d_name.name);
+   EXT4_ERROR_INODE(dir, "'%pd' linked to parent dir",
+      dentry);
    return ERR_PTR(-EIO);
   }
   inode = ext4_iget(dir->i_sb, ino);
-- 
1.7.1